src/interface.jl

```diff
@@ -0,0 +1,252 @@
+
+# default `forward` implementations; should in general implement efficient way
+# of computing both `transform` and `logabsdetjac` together.
+forward(b::Bijector, x) = (rv=transform(b, x), logabsdetjac=logabsdetjac(b, x))
+forward(ib::Inversed{<: Bijector}, y) = (rv=transform(ib, y), logabsdetjac=logabsdetjac(ib, y))
+
+# defaults implementation for inverses
+logabsdetjac(ib::Inversed{<: Bijector}, y) = - logabsdetjac(ib.orig, transform(ib, y))
+
+inv(b::Bijector) = Inversed(b)
+inv(ib::Inversed{<:Bijector}) = ib.orig
+
+# AD implementations
+function jacobian(b::ADBijector{<: ForwardDiffAD}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
+end
+function jacobian(b::Inversed{<: ADBijector{<: ForwardDiffAD}}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
+end
+function jacobian(b::Inversed{<: ADBijector{<: ForwardDiffAD}}, y::AbstractVector{<: Real})
+    ForwardDiff.jacobian(z -> transform(b, z), y)
+end
+
+function jacobian(b::ADBijector{<: TrackerAD}, y::Real)
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::Real)
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::AbstractVector{<: Real})
+    Tracker.jacobian(z -> transform(b, z), y)
+end
+
+# TODO: allow batch-computation, especially for univariate case?
+"Computes the absolute determinant of the Jacobian of the inverse-transformation."
+function logabsdetjac(b::ADBijector, x::Real)
+    log(abs(jacobian(b, x)))
+end
+function logabsdetjac(b::ADBijector, x::AbstractVector{<:Real})
+    fact = lu(jacobian(b, x), check=false)
+    issuccess(fact) ? log(abs(det(fact))) : -Inf # TODO: or smallest possible float?
+end
+
+###############
+# Composition #
+###############
+
+struct Composed{A} <: Bijector
+    ts::A
+end
+
+compose(ts...) = Composed(ts)
+
+# The transformation of `Composed` applies functions left-to-right
+# but in mathematics we usually go from right-to-left; this reversal ensures that
+# when we use the mathematical composition ∘ we get the expected behavior.
+# TODO: change behavior of `transform` of `Composed`?
+∘(b1::B1, b2::B2) where {B1 <: Bijector, B2 <: Bijector} = compose(b2, b1)
+
+inv(ct::Composed) = Composed(map(inv, reverse(ct.ts)))
+
+# TODO: can we implement this recursively, and with aggressive inlining, make this type-stable?
+function transform(cb::Composed, x)
+    res = x
+    for b ∈ cb.ts
+        res = transform(b, res)
+    end
+
+    return res
+end
+
+(cb::Composed)(x) = transform(cb, x)
+
+function forward(cb::Composed, x)
+    res = (rv=x, logabsdetjac=0)
+    for t in cb.ts
+        res′ = forward(t, res.rv)
+        res = (rv=res′.rv, logabsdetjac=res.logabsdetjac + res′.logabsdetjac)
+    end
+    return res
+end
+
+##############################
+# Example bijector: Identity #
+##############################
+
+struct Identity <: Bijector end
```

Comment:
Do we actually need this Bijector for anything? If not, could probably just get rid of it (even if it is nice to have in principle)

Reply:
It's use to have as a default, or at the very least, return for distributions like `Normal` and so on.
The only other alternative I can think of is to define `transform` and so on for `Base.identity`, but this is of course not `<: Bijector` and so we lose the ability to dispatch on bijectors only.

So I actually think it's necessary. We can remove it in the future, if we realize it's redundant?:)